File: admin-panel/src/services/jobService.ts
import { apiClient } from './api';
import { ApiResponse, Job, PaginatedResponse, JobFormData } from '@/types';

export const jobService = {
  getAll: async (token: string, params?: { 
    page?: number; 
    limit?: number; 
    search?: string;
    category?: string;
  }) => {
    const queryParams = new URLSearchParams();
    if (params?.page) queryParams.append('page', params.page.toString());
    if (params?.limit) queryParams.append('limit', params.limit.toString());
    if (params?.search) queryParams.append('search', params.search);
    if (params?.category) queryParams.append('category', params.category);
    
    const query = queryParams.toString() ? `?${queryParams.toString()}` : '';
    return apiClient.get<PaginatedResponse<Job>>(`/admin/jobs/${query}`, token);
  },

  getById: async (token: string, id: string) => {
    return apiClient.get<Job>(`/admin/jobs/${id}`, token);
  },

  create: async (token: string, data: JobFormData) => {
    return apiClient.post<ApiResponse<Job>>('/admin/jobs/add', data, token);
  },

  update: async (token: string, id: string, data: Partial<JobFormData>) => {
    return apiClient.put<ApiResponse<Job>>(`/admin/jobs/${id}`, data, token);
  },

  delete: async (token: string, id: string) => {
    return apiClient.delete<ApiResponse<null>>(`/admin/jobs/${id}`, token);
  },

  getCategories: async () => {
    return apiClient.get<string[]>('/jobs/categories/list');
  },

  getStats: async (token: string) => {
    return apiClient.get<{ total: number }>(`/admin/jobs/stats/count`, token);
  },
};

File: workscout/lib/services/job_service.dart
import 'package:workscout/data/model/job_model.dart';
import 'package:workscout/services/api_client.dart';

class JobService {
  Future<List<Job>> fetchJobs({String? category, String? search, int page = 1, int limit = 20}) async {
    final queryParams = <String, String>{
      'page': page.toString(),
      'limit': limit.toString(),
    };
    if (category != null && category.isNotEmpty) queryParams['category'] = category;
    if (search != null && search.isNotEmpty) queryParams['search'] = search;

    final response = await ApiClient.get('/jobs', queryParams: queryParams);
    final List<dynamic> jsonList = response['jobs'] as List<dynamic>;
    return jsonList.map((json) => Job.fromJson(json as Map<String, dynamic>)).toList();
  }

  Future<Job> fetchJobById(String id) async {
    final response = await ApiClient.get('/jobs/$id');
    return Job.fromJson(response as Map<String, dynamic>);
  }
}
